api/controllers/install/controller.go

```diff
@@ -88,6 +88,10 @@ func (c *InstallController) SetConfig(ctx context.Context, config *types.Install
 }
 
 func (c *InstallController) SetStatus(ctx context.Context, status *types.InstallationStatus) error {
+	if err := c.installationManager.ValidateStatus(status); err != nil {
+		return fmt.Errorf("validate: %w", err)
+	}
```

Comment:
I'll be doing some follow up PRs that test the validations on the installationManager